packages/backend/src/index.ts

```diff
@@ -75,11 +76,23 @@ app.get('*', (req, res) => {
 });
 
 // errors
-app.use(
-    async (error: Error, req: Request, res: Response, next: NextFunction) => {
-        await errorHandler(error, res);
-    },
-);
+app.use((error: Error, req: Request, res: Response) => {
```

Comment:
Does this alter the behaviour of this middleware?

Express docs say that the ErrorHandler must be a 4 argument middleware (I guess if an error happens early then normal middleware are skipped but this should still run): https://expressjs.com/en/guide/error-handling.html (see "Writing error handlers").

Optionally: could split out the error parsing middleware (creating LightdashError) vs. the analytics middleware

Reply:
nice catch :)